<!DOCTYPE html>
<html lang="zh-CN">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>
        // function fetchData() {
        //     return new Promise((resolve, reject) => {
        //         setTimeout(() => {
        //             let data = { name: "张三", age: 30 };
        //             resolve(data); // 假设请求成功
        //         }, 2000); // 模拟2秒的延迟
        //     });
        // }

        // fetchData()
        //     .then(data => {
        //         console.log("获取数据成功:", data);
        //     })
        //     .catch(error => {
        //         console.log("获取数据失败:", error);
        //     });


        // function step1() {
        //     return new Promise((resolve, reject) => {
        //         setTimeout(() => {
        //             console.log("步骤 1 完成");
        //             resolve("结果 1");
        //         }, 1000);
        //     });
        // }

        // function step2(result1) {
        //     return new Promise((resolve, reject) => {
        //         setTimeout(() => {
        //             console.log("步骤 2 完成，接收到的结果是:", result1);
        //             resolve("结果 2");
        //         }, 1000);
        //     });
        // }

        // step1()
        //     .then(result1 => step2(result1))  // 将结果传递到下一步
        //     .then(result2 => {
        //         console.log("步骤 3 完成，接收到的结果是:", result2);
        //     })
        //     .catch(error => {
        //         console.error("发生错误:", error);
        //     });


        // let promise1 = new Promise((resolve, reject) => {
        //     setTimeout(() => resolve("数据 1"), 1000);
        // });

        // let promise2 = new Promise((resolve, reject) => {
        //     setTimeout(() => resolve("数据 2"), 1500);
        // });

        // let promise3 = new Promise((resolve, reject) => {
        //     setTimeout(() => resolve("数据 3"), 2000);
        // });

        // Promise.all([promise1, promise2, promise3])
        //     .then(results => {
        //         console.log("所有异步操作完成:", results);
        //     })
        //     .catch(error => {
        //         console.error("某个操作失败:", error);
        //     });

        let promise = new Promise((resolve, reject) => {
            resolve("成功");
            // reject("失败");
        }).then(result => {
            console.log("result:", result)
        }).catch(error => {
            console.log("error:", error)
        }).finally(() => {
            console.log("异步操作完成");
        })

    </script>
</body>

</html>